src/double_linked_list.py

```diff
@@ -46,47 +43,67 @@ def append(self, val):
         self._counter += 1
 
     def pop(self):
-        """Remove & return the value of the head of the Linked List."""
+        """Remove & return the value of the head of the DLL."""
         if not self.head:
             raise IndexError("The list is empty, so there's nothing to pop.")
-        output = self.head.data
-        self.head.next_node.previous = None
-        self.head = self.head.next_node
-        self._counter -= 1
-        return output
+        elif self.head == self.tail:
+            output = self.head
+            self.head = None
+            self.tail = None
+            self._counter -= 1
+            return output.data
+        else:
+            output = self.head.data
+            self.head.next_node.previous = None
+            self.head = self.head.next_node
+            self._counter -= 1
+            return output
```

Comment:
Try to be consistent in what `output` is. In the `elif`, `output` is holding the whole node. In the `else`, it's just the data on the node. Neither method is alone wrong, just make a choice and stick to that choice.